db/migrate/20150930183738_migrate_content_hosts.rb

```diff
@@ -0,0 +1,213 @@
+
+    begin
+      logger.info("Removing Candlepin consumer #{system.uuid}")
+      Katello::Resources::Candlepin::Consumer.destroy(system.uuid)
+    rescue RestClient::Exception => e
+      logger.warn("Exception when destroying candlepin consumer #{system.uuid}:#{e.inspect}")
+    end
+
+    begin
+      logger.info("Removing Pulp consumer #{system.uuid}")
+      Katello.pulp_server.extensions.consumer.delete(system.uuid)
+    rescue RestClient::ResourceNotFound
+      logger.warn("Pulp consumer not found for consumer #{system.uuid} proceeding.")
+      #do nothing
+    rescue RestClient::Exception => e
+      logger.warn("Exception when destroying pulp consumer #{system.uuid}:#{e.inspect}")
+    end
+
+    logger.info("Removing system #{system.uuid}")
+    system.destroy
+  end
+
+  def unregister_all_but_last_system(systems)
+    systems_by_created_date = systems.sort_by(&:created_at)
+    system = systems_by_created_date.pop
+
+    systems_by_created_date.each do |system_to_remove|
+      unregister_system(system_to_remove)
+    end
+
+    system
+  end
+
+  # rubocop:disable MethodLength
+  def up
+    if User.where(:login => User::ANONYMOUS_API_ADMIN).first.nil?
+      logger.warn("Foreman anonymous admin does not exist, skipping content host migration.")
+      return
+    end
+
+    User.current = User.anonymous_api_admin
+
+    ensure_one_system_per_hostname(Katello::System.all)
+
+    systems = get_systems_with_facts(Katello::System.all)
+
+    systems.each do |system|
+      hostname = system.facts['network.hostname']
+
+      logger.info("Processing content host #{system.uuid} #{hostname}")
+
+      if hostname.nil?
+        logger.warn("Content host #{system.uuid} does not have a hostname, removing.")
+        unregister_system(system)
+        break
+      end
+
+      hosts = ::Host.where(:name => hostname)
+      if hosts.empty? # no host exists
+        logger.info("No host exists with hostname #{hostname}, creating new host.")
+        params = system.attributes.to_options
+        params[:facts] = system.facts
+        host = Katello::Host::SubscriptionAspect.new_host_from_rhsm_params(params, system.environment.organization, Location.default_location)
+        host.save!
+
+        create_content_aspect(host, system)
+        create_subscription_aspect(host, system)
+
+      elsif hosts.where(:organization_id => system.environment.organization.id).empty? # host is not in the correct org
+        logger.warn("Found host with hostname #{hostname} but it's in org #{hosts[0].org.name} instead of #{system.environment.organization.name}.")
+        host = hosts.first
+
+        create_content_aspect(host, system)
+        unregister_system(system)
+
+      else #host exists in the correct org
+        logger.info("Found host with hostname #{hostname}.")
+        host = hosts.first
+
+        host.content_aspect = create_content_aspect(host, system)
+        host.subscription_aspect = create_subscription_aspect(host, system)
```

Comment:
Both create_content_aspect and create_subscription_aspect aren't returning their actual objects, but the results from save!  which would be true (unless an exception is thrown).